### PR TITLE
Add Vue and React examples for the Calendar

### PR DESCRIPTION
**Pull request by DokaRus**

### CODE REVIEW

**Comment by DokaRus** [comment]
I think we should update [this line](https://github.com/DokaRus/devextreme-documentation/blame/calendar_examples_20_1/concepts/05%20Widgets/Calendar/15%20Customize%20Cell%20Appearance.md#L1) to match the examples below - Angular uses `dxTemplate` element, Vue uses `template` element, React uses `cellRender` function.

Now she looks like this:

>For Angular, AngularJS, and Knockout apps, DevExtreme provides the [dxTemplate](/api-reference/10%20UI%20Widgets/Markup%20Components/dxTemplate '/Documentation/ApiReference/UI_Widgets/Markup_Components/dxTemplate/') markup component. The following code shows how to use **dxTemplate** to define templates for cells.

**Comment by AlexJameson** [comment]
> 
> 
> I think we should update [this line](https://github.com/DokaRus/devextreme-documentation/blame/calendar_examples_20_1/concepts/05%20Widgets/Calendar/15%20Customize%20Cell%20Appearance.md#L1) to match the examples below - Angular uses `dxTemplate` element, Vue uses `template` element, React uses `cellRender` function.
> 
> Now she looks like this:
> 
> > For Angular, AngularJS, and Knockout apps, DevExtreme provides the [dxTemplate](/api-reference/10%20UI%20Widgets/Markup%20Components/dxTemplate) markup component. The following code shows how to use **dxTemplate** to define templates for cells.

I agree, we should. I've added this task to my backlog.

**Review by AlexJameson — APPROVED** [review]

File: concepts/05 Widgets/Calendar/05 Handle the Value Change Event.md [hunk 1]
```diff
@@ -39,6 +37,48 @@ To process a new calendar value, you need to handle the value change event. If t
         // ...
     })
 
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template> 
+        <DxCalendar @value-changed="calendarValueChanged" />
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Calendar/05 Handle the Value Change Event.md]: It looks like in Angular and Vue we should show two-way binding and in React state changing.

File: concepts/05 Widgets/Calendar/00 Overview.md [hunk 2]
```diff
@@ -40,6 +40,46 @@ The following code adds a simple **Calendar** to your page.
...
 
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template> 
+        <DxCalendar :value="date" />
+    </template>
+
+    <script>
+    import DxCalendar from 'devextreme-vue/calendar';
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Calendar/00 Overview.md:54]: Please do not forget to import CSS.

File: concepts/05 Widgets/Calendar/12 Specify the Value Range.md [hunk 3]
```diff
@@ -37,6 +37,57 @@ Use the [min](/api-reference/10%20UI%20Widgets/dxCalendar/1%20Configuration/min.
...
+    import DxCalendar from 'devextreme-vue/calendar';
+
+    export default {
+        components: {
+            DxCalendar
+        },
+        data() {
+            return {
+                date: new Date(),
+                minDate: Date = new Date(2000, 1, 1),
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Calendar/12 Specify the Value Range.md]:
```suggestion
                minDate: new Date(2000, 1, 1),
```

File: concepts/05 Widgets/Calendar/12 Specify the Value Range.md [hunk 4]
```diff
@@ -37,6 +37,57 @@ Use the [min](/api-reference/10%20UI%20Widgets/dxCalendar/1%20Configuration/min.
...
+                maxDate: Date = new Date(2020, 12, 31)
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Calendar/12 Specify the Value Range.md]:
```suggestion
                maxDate: new Date(2020, 12, 31)
```

File: concepts/05 Widgets/Calendar/20 Keyboard Support.md [hunk 5]
```diff
@@ -73,6 +73,58 @@ You can implement a custom handler for a key using the [registerKeyHandler(key,
         // ...
     })
 
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template> 
+        <DxCalendar @initialized="registerKeyHandlers" />
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Calendar/20 Keyboard Support.md]: We usually call registerKeyHandler on `mounted`/`componentDidMount()`:
https://js.devexpress.com/Documentation/20_1/Guide/Widgets/Button/Keyboard_Support/

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

File: concepts/05 Widgets/Calendar/05 Handle the Value Change Event.md (hunk 1 above)
Comment (DokaRus) [reply concepts/05 Widgets/Calendar/05 Handle the Value Change Event.md]: Is it actual for the **"Handle the Value Change Event"** topic?
During this event we can not only synchronize the values[two-way binding], but also perform other actions - load the data source, call specific function etc.

Comment (RomanTsukanov) [reply concepts/05 Widgets/Calendar/05 Handle the Value Change Event.md]: Can we have both two-way binding and the valueChanged event?

Comment (DokaRus) [reply concepts/05 Widgets/Calendar/05 Handle the Value Change Event.md]: Yep. Topic updated

File: concepts/05 Widgets/Calendar/05 Handle the Value Change Event.md [hunk 6]
```diff
@@ -39,6 +37,73 @@ To process a new calendar value, you need to handle the value change event. If t
...
+    import React from 'react';
+
+    import 'devextreme/dist/css/dx.common.css';
+    import 'devextreme/dist/css/dx.light.css';
+
+    import Calendar from 'devextreme-react/calendar';
+
+    class App extends React.Component {
+        constructor() {
+            this.super();
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Calendar/05 Handle the Value Change Event.md]:
```suggestion
            super();
```

File: concepts/05 Widgets/Calendar/05 Handle the Value Change Event.md [hunk 7]
```diff
@@ -39,6 +37,73 @@ To process a new calendar value, you need to handle the value change event. If t
...
+
+    export default {
+        components: {
+            DxCalendar
+        },
+        data() {
+            calendarValue: new Date()
+        },
+        methods: {
+            calendarValueChanged: function({ previousValue, value: newValue }) {
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Calendar/05 Handle the Value Change Event.md]: I know that destructuring is all the rage, but these are tutorial examples. Let's keep them as clear as possible, without shortcuts that not everyone know.

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

File: concepts/05 Widgets/Calendar/05 Handle the Value Change Event.md (hunk 7 above)
Comment (DokaRus) [reply concepts/05 Widgets/Calendar/05 Handle the Value Change Event.md]: got it

**Review by RomanTsukanov — APPROVED** [review]
👍

**Pull request merged**